ipywidgets/widgets/widget.py

```diff
@@ -717,9 +722,20 @@ def _ipython_display_(self, **kwargs):
                     'model_id': self._model_id
                 }
             }
-            display(data, raw=True)
+            return data
 
-            self._handle_displayed(**kwargs)
+    def _ipython_display_(self, **kwargs):
+        """Called when `IPython.display.display` is called on a widget.
+
+        Note: if we are in IPython 6.1 or later, we return NotImplemented so
+        that _repr_mimebundle_ is used directly.
+        """
+        if ipython_version_info >= (6, 1):
+            raise NotImplementedError
```

Comment:
I guess this should be `raise NotImplemented`?

Reply:
No, `NotImplemented` is not an error class, so raising that will throw a `TypeError`.

Reply:
(compare methods *return* `NotImplemented`, while subclasses with undefined methods *raise* `NotImplementedError`: https://docs.python.org/3.6/library/exceptions.html#NotImplementedError, https://docs.python.org/3.6/library/constants.html#NotImplemented)

Reply:
or do what @SylvainCorlay said and just put the definition into an if statement.